eis_qgis_plugin/processing/algorithms/raster_processing/reclassify_with_defined_intervals.py

```diff
@@ -0,0 +1,57 @@
+from qgis.core import (
+    QgsProcessingParameterBand,
+    QgsProcessingParameterNumber,
+    QgsProcessingParameterRasterDestination,
+    QgsProcessingParameterRasterLayer,
+)
+
+from eis_qgis_plugin.processing.eis_processing_algorithm import EISProcessingAlgorithm
+
+
+class EISReclassifyWithDefinedIntervals(EISProcessingAlgorithm):
+    def __init__(self) -> None:
+        super().__init__()
+
+        self._name = "reclassify_with_defined_intervals"
+        self._display_name = "Reclassify with defined intervals"
+        self._group = "Raster Processing"
+        self._group_id = "raster_processing"
+        self._short_help_string = (
+            "Reclassify raster data set with defined intervals."
+        )
+
+    def initAlgorithm(self, config=None):
+        self.alg_parameters = [
+            "input_raster",
+            "interval_size",
+            "bands",
+            "output_raster",
+        ]
+
+        input_raster_param = QgsProcessingParameterRasterLayer(
+            name=self.alg_parameters[0], description="Input raster"
+        )
+        input_raster_param.setHelp("The input raster data set.")
+        self.addParameter(input_raster_param)
+
+        interval_size_param = QgsProcessingParameterNumber(
+            name=self.alg_parameters[1],
+            description="Interval size",
+            minValue=1,
+        )
+        interval_size_param.setHelp("The interval size for Defined intervals.")
+        self.addParameter(interval_size_param)
+
+        bands_param = QgsProcessingParameterBand(
+            name=self.alg_parameters[2],
+            description="Raster bands",
+            parentLayerParameterName=self.alg_parameters[0],
+        )
+        bands_param.setHelp("The bands to be reclassified.")
```

Comment:
As a very minor nitpick, I would change the description and help of band to singlar form since the user can select only 1 band. This would apply for all algs.
```suggestion
            description="Raster band",
            parentLayerParameterName=self.alg_parameters[0],
        )
        bands_param.setHelp("The band to be reclassified.")
```

Reply:
I think that is an important change that clarifies the implementation. Is it ok to commit this via GitHub?

Reply:
Better to commit maybe yourself, at least now that I did not mark all of the modifications.